src/amo/reducers/home.js

```diff
@@ -115,12 +104,12 @@ const reducer = (
       };
 
     case LOAD_HOME_ADDONS: {
-      const {
-        collections,
-        featuredExtensions,
-        featuredThemes,
-        popularExtensions,
-      } = action.payload;
+      const { collections, shelves } = action.payload;
+
+      const shelvesToLoad = {};
+      Object.keys(shelves).forEach((shelfName) => {
+        shelvesToLoad[shelfName] = createInternalAddons(shelves[shelfName]);
+      });
```

Comment:
You could also use `reduce()`, but that works too

```js
const shelvesToLoad = Object.keys(shelves).reduce((shelvesToLoad, shelfName) => {
  return {
    ...shelvesToLoad,
    [shelfName]: createInternalAddons(shelves[shelfName]),
  };
}, {});
```

Reply:
Thanks, I don't often think to use `reduce` but it's pretty nice here.